crates/bitwarden-crypto/src/safe/password_protected_key_envelope.rs

```diff
@@ -0,0 +1,572 @@
+}
+
+impl Argon2RawSettings {
+    /// Creates default Argon2 settings based on the platform. This currently is a static preset
+    /// based on the target os
+    fn default_for_platform() -> Self {
+        // iOS has memory limitations in the auto-fill context. So, the memory is halved
+        // but the iterations are doubled
+        if cfg!(target_os = "ios") {
+            // The SECOND RECOMMENDED option from: https://datatracker.ietf.org/doc/rfc9106/, with halved memory and doubled iteration count
+            Self {
+                iterations: 6,
+                memory: 32 * 1024, // 32 MiB
+                parallelism: 4,
+                salt: make_salt(),
+            }
+        } else {
+            // The SECOND RECOMMENDED option from: https://datatracker.ietf.org/doc/rfc9106/
+            // The FIRST RECOMMENDED option currently still has too much memory consumption for most
+            // clients except desktop.
+            Self {
+                iterations: 3,
+                memory: 64 * 1024, // 64 MiB
+                parallelism: 4,
+                salt: make_salt(),
+            }
+        }
+    }
+}
+
+impl From<&Argon2RawSettings> for Header {
+    fn from(settings: &Argon2RawSettings) -> Header {
+        let builder = HeaderBuilder::new()
+            .value(ARGON2_ITERATIONS, Integer::from(settings.iterations).into())
+            .value(ARGON2_MEMORY, Integer::from(settings.memory).into())
+            .value(
+                ARGON2_PARALLELISM,
+                Integer::from(settings.parallelism).into(),
+            )
+            .value(ARGON2_SALT, Value::from(settings.salt.to_vec()));
+
+        let mut header = builder.build();
+        header.alg = Some(coset::Algorithm::PrivateUse(ALG_ARGON2ID13));
+        header
+    }
+}
+
+impl TryInto<Params> for &Argon2RawSettings {
+    type Error = PasswordProtectedKeyEnvelopeError;
+
+    fn try_into(self) -> Result<Params, PasswordProtectedKeyEnvelopeError> {
+        Params::new(self.memory, self.iterations, self.parallelism, Some(32))
+            .map_err(|_| PasswordProtectedKeyEnvelopeError::KdfError)
+    }
+}
+
+impl TryInto<Argon2RawSettings> for &Header {
+    type Error = PasswordProtectedKeyEnvelopeError;
+
+    fn try_into(self) -> Result<Argon2RawSettings, PasswordProtectedKeyEnvelopeError> {
+        Ok(Argon2RawSettings {
+            iterations: extract_integer(self, ARGON2_ITERATIONS, "iterations")?.try_into()?,
+            memory: extract_integer(self, ARGON2_MEMORY, "memory")?.try_into()?,
+            parallelism: extract_integer(self, ARGON2_PARALLELISM, "parallelism")?.try_into()?,
+            salt: extract_bytes(self, ARGON2_SALT, "salt")?
+                .try_into()
+                .map_err(|_| {
+                    PasswordProtectedKeyEnvelopeError::ParsingError(
+                        "Invalid Argon2 salt".to_string(),
+                    )
+                })?,
+        })
+    }
+}
+
+fn make_salt() -> [u8; 32] {
```

Comment:
We have `32` in a few places as the fixed salt size. Consider making a constant instead of "magic numbers".

Reply:
Replaced all the magic numbers and moved the constants up in the file.